### ISSUE TITLE
Failed to load dynamic library 'libc.so.6': dlopen failed: library "libc.so.6" not found

### ISSUE THREAD

**Issue by ksh-b** [brendan-duncan/archive#361]
Platform: Android 11
archive version: 4.0.1

Steps to reproduce:
Try to use `extractFileToDisk` method for extracting zip file.

This line seems to be causing the issue
[`posix.isPosixSupported`](https://github.com/brendan-duncan/archive/blob/main/lib/src/io/extract_archive_to_disk.dart#L237)

from pubspec.lock:
```
...
  archive:
    dependency: "direct main"
    description:
      name: archive
      sha256: "08064924cbf0ab88280a0c3f60db9dd24fec693927e725ecb176f16c629d1cb8"
      url: "https://pub.dev"
    source: hosted
    version: "4.0.1"
...
  posix:
    dependency: transitive
    description:
      name: posix
      sha256: a0117dc2167805aa9125b82eee515cc891819bac2f538c83646d355b16f58b9a
      url: "https://pub.dev"
    source: hosted
    version: "6.0.1"
```

The method was working fine in archive version 3.6.1

**Comment by ksh-b** [comment]
maybe this happens only for some devices
https://github.com/onepub-dev/dart_posix/issues/4

**Comment by CaptainDario** [comment]
This happens to me too when running on an emulator.

**Comment by brendan-duncan** [comment]
Thanks for the report, I'll see what I can do about it

**Comment by brendan-duncan** [comment]
I put the posix check into an exception catcher to keep it from crashing. It will just not be able to chmod the file to the archive file permissions in that case, which is how it was in 3.x.

**Comment by josercc** [comment]
> I put the posix check into an exception catcher to keep it from crashing. It will just not be able to chmod the file to the archive file permissions in that case, which is how it was in 3.x.

I am currently using version 4.0 and still receive this error message

ArgumentError: Invalid argument(s): Failed to load dynamic library 'libc.so.6': dlopen failed: library "libc.so.6" not found
  #0      _open (dart:ffi-patch/ffi_dynamic_library_patch.dart:11)
  #1      new DynamicLibrary.open (dart:ffi-patch/ffi_dynamic_library_patch.dart:22)
  #2      new Libc._internal (package:posix/src/libc.dart:25)
  #3      Libc._self (package:posix/src/libc.dart:27)
  #4      isPosixSupported (package:posix/posix.dart)
  #5      extractFileToDisk (package:archive/src/io/extract_archive_to_disk.dart:237)
  #6      extractFile (package:meta_winner_app/app/modules/download_progress/controllers/download_unity_asset_bridge.dart:220)
  File "<asynchronous suspension>"
  #8      _RemoteRunner._run (dart:isolate:1092)

**Comment by brendan-duncan** [comment]
The update is in git, I haven't published it yet. I'll try to do that soon.

**Comment by noobar** [comment]
I confirmed this issue can be fixed by updating v4.0.1 to v4.0.2.
My devices are Pixel8a (Android 15) and Lenovo Tab M8 4th (Android 13), both reproduced the issue in v4.0.1.
Thanks!
